### PR TITLE
Azure AD reporting: Adding defaultMethod to Auth Methods report and sessionControlsNotSatisfied to applied CA 

### PR DESCRIPTION
**Pull request by besiler**
Added

- SessionControlNotSatisified to appliedConditionalAccessPolicy, schema PR 5969532
- defaultMethod to auth method report, schema PR 5969412

### CODE REVIEW

File: api-reference/beta/resources/userregistrationdetails.md [hunk 1]
```diff
@@ -26,6 +26,7 @@ Inherits from [entity](../resources/entity.md).
 ## Properties
 |Property|Type|Description|
 |:---|:---|:---|
+|defaultMethod|String|The method the user or admin selected as default for performing multi-factor authentication for the user.|
```
Comment (FaithOmbongi) [inline api-reference/beta/resources/userregistrationdetails.md:29]: Is it only for MFA?

Comment (FaithOmbongi) [inline api-reference/beta/resources/userregistrationdetails.md:29]: Is it only for MFA?

File: api-reference/beta/resources/userregistrationdetails.md [hunk 2]
```diff
@@ -26,6 +26,7 @@ Inherits from [entity](../resources/entity.md).
...
 |id|String|User object identifier in Azure AD. Inherited from [entity](../resources/entity.md).|
 |isMfaCapable|Boolean|Whether the user has registered a strong authentication method for multi-factor authentication. The method must be allowed by the [authentication methods policy](../resources/authenticationmethodspolicy.md). Supports `$filter` (`eq`).|
 |isMfaRegistered|Boolean|Whether the user has registered a strong authentication method for multi-factor authentication. The method may not necessarily be allowed by the [authentication methods policy](../resources/authenticationmethodspolicy.md).  Supports `$filter` (`eq`).|
```
Comment (FaithOmbongi) [inline api-reference/beta/resources/userregistrationdetails.md:32]: Please add the method to the JSON representation

File: api-reference/beta/resources/appliedconditionalaccesspolicy.md [hunk 3]
```diff
@@ -29,6 +29,7 @@ Indicates the attributes related to applied conditional access policy or policie
...
 |result|appliedConditionalAccessPolicyResult| Indicates the result of the CA policy that was triggered. Possible values are: `success`, `failure`, `notApplied` (Policy isn't applied because policy conditions were not met),`notEnabled` (This is due to the policy in disabled state), `unknown`, `unknownFutureValue`, `reportOnlySuccess`, `reportOnlyFailure`, `reportOnlyNotApplied`, `reportOnlyInterrupted`. Note that you must use the `Prefer: include-unknown-enum-members` request header to get the following values in this [evolvable enum](/graph/best-practices-concept#handling-future-members-in-evolvable-enumerations): `reportOnlySuccess`, `reportOnlyFailure`, `reportOnlyNotApplied`, `reportOnlyInterrupted`.|
+|sessionControlsNotSatisfied|String collection|Refers to the session controls that a sign-in activity did not satisfy. (Example: “Application enforced Restrictions”).||
```
Comment (FaithOmbongi) [inline api-reference/beta/resources/appliedconditionalaccesspolicy.md:32]: Please also ad this new property to the JSON representation
```suggestion
|sessionControlsNotSatisfied|String collection|Refers to the session controls that a sign-in activity did not satisfy. (Example: `Application enforced Restrictions`).||
```

File: changelog/Microsoft.AAD.Reporting.json [hunk 4]
```diff
@@ -1,5 +1,41 @@
...
+          "ChangeType": "Addition",
+          "Description": "Added the **sessionControlsNotSatisfied** property to [appliedConditionalAccessPolicy](https://docs.microsoft.com/en-us/graph/api/resources/appliedConditionalAccessPolicy?view=graph-rest-beta) resource.",
+          "Target": "appliedConditionalAccessPolicy"
+        }
+      ],
+      "Id": "3e4f113d-2b6b-4bb1-9cd7-24fa265f5e2a",
+      "Cloud": "Prod",
+      "Version": "beta",
+      "CreatedDateTime": "2022-04-18T23:23:47.3148004Z",
+      "WorkloadArea": "Reports and audit",
```
Comment (FaithOmbongi) [inline changelog/Microsoft.AAD.Reporting.json:18]: Minor fixes
```suggestion
          "Description": "Added the **sessionControlsNotSatisfied** property to the [appliedConditionalAccessPolicy](https://docs.microsoft.com/en-us/graph/api/resources/appliedConditionalAccessPolicy?view=graph-rest-beta) resource.",
          "Target": "appliedConditionalAccessPolicy"
        }
      ],
      "Id": "3e4f113d-2b6b-4bb1-9cd7-24fa265f5e2a",
      "Cloud": "Prod",
      "Version": "beta",
      "CreatedDateTime": "2022-04-18T23:23:47.3148004Z",
      "WorkloadArea": "Reports",
```

File: changelog/Microsoft.AAD.Reporting.json [hunk 5]
```diff
@@ -1,5 +1,41 @@
...
+          "ChangeType": "Addition",
+          "Description": "Added the **defaultMethod** property to [userRegistrationDetails](https://docs.microsoft.com/en-us/graph/api/resources/userRegistrationDetails?view=graph-rest-beta) resource.",
+          "Target": "userRegistrationDetails"
+        }
+      ],
+      "Id": "5527244c-2c92-4dea-ba34-b18e3a7c3ba2",
+      "Cloud": "Prod",
+      "Version": "beta",
+      "CreatedDateTime": "2022-04-18T21:35:40.9738581Z",
+      "WorkloadArea": "Reports and audit",
```
Comment (FaithOmbongi) [inline changelog/Microsoft.AAD.Reporting.json:36]: Minor fixes
```suggestion
          "Description": "Added the **defaultMethod** property to the [userRegistrationDetails](https://docs.microsoft.com/en-us/graph/api/resources/userRegistrationDetails?view=graph-rest-beta) resource.",
          "Target": "userRegistrationDetails"
        }
      ],
      "Id": "5527244c-2c92-4dea-ba34-b18e3a7c3ba2",
      "Cloud": "Prod",
      "Version": "beta",
      "CreatedDateTime": "2022-04-18T21:35:40.9738581Z",
      "WorkloadArea": "Reports",
```

**Review by FaithOmbongi — CHANGES_REQUESTED** [review]
Thank you @besiler for your work on this PR. A few fixes plus add the two new properties to the respective JSON representation sections.

File: api-reference/beta/resources/userregistrationdetails.md (hunk 1 above)
Comment (besiler) [reply api-reference/beta/resources/userregistrationdetails.md:29]: Yes, I think the concept of "default" only applies to MFA, even though the auth methods are used more broadly. The MFA team refers to it that way in the existing Graph docs: https://docs.microsoft.com/en-us/graph/api/resources/authenticationmethods-overview?view=graph-rest-1.0

image

File: api-reference/beta/resources/userregistrationdetails.md (hunk 2 above)
Comment (besiler) [reply api-reference/beta/resources/userregistrationdetails.md:32]: Added

**Comment by besiler** [comment]
@FaithOmbongi resolved the questions/suggestions above, but the Graph review team has asked us to do defaultMethod as an enum instead. I'll make that change once the devs have a new PR with changelog etc. 

For now, we should keep this PR marked as not ready to merge please

**Comment by besiler** [comment]
@FaithOmbongi I updated defaultMfaMethod to an enum based on guidance from the Graph architects. This should be ready to merge if you don't have further comments

**Review by FaithOmbongi — APPROVED** [review]
Hi @besiler - Changes look good. Added one suggestion to resolve the Build warning. Otherwise, approved.

**Pull request merged**